File: 3way/problem7/solution.c
char* reverseVowels(char* s) {
    char isVowel[128];
    char vowels[] = {'a','A','e','E','u','U','i','I','o','O'};
    int start = 0;
    int end = strlen(s) - 1;

    for(int i = 0; i < 10; i++){
        isVowel[vowels[i]] = true;
    }

    while(start < end){
        if(isVowel[s[start]]){
            while(true){
                if(isVowel[s[end]]){
                    char c = s[start];
                    s[start] = s[end];
                    s[end] = c;
                    end--;
                    break;
                }
                end--;
            }
        }
        start++;
    }

    return s;

}

File: 3way/problem7/solution.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {

    const vowels = new Set(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']);
    const chars = s.split('');
    let start = 0;
    let end = s.length - 1;

    while (start < end) {
        while (start < end && !vowels.has(chars[start])) {
            start++;
        }
        while (start < end && !vowels.has(chars[end])) {
            end--;
        }
        [chars[start], chars[end]] = [chars[end], chars[start]];
        start++;
        end--;
    }

    return chars.join('');

};
